ext/src/http/client/curl/http_client_curl.cc

```diff
@@ -3,16 +3,577 @@
 
 #include "opentelemetry/ext/http/client/curl/http_client_curl.h"
 
-bool opentelemetry::ext::http::client::curl::Session::CancelSession() noexcept
+#include <list>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext
+{
+namespace http
+{
+namespace client
+{
+namespace curl
+{
+
+HttpCurlGlobalInitializer::HttpCurlGlobalInitializer()
+{
+  curl_global_init(CURL_GLOBAL_ALL);
+}
+
+HttpCurlGlobalInitializer::~HttpCurlGlobalInitializer()
+{
+  curl_global_cleanup();
+}
+
+nostd::shared_ptr<HttpCurlGlobalInitializer> HttpCurlGlobalInitializer::GetInstance()
+{
+  static nostd::shared_ptr<HttpCurlGlobalInitializer> shared_initializer{
+      new HttpCurlGlobalInitializer()};
+  return shared_initializer;
+}
+
+void Session::SendRequest(
+    std::shared_ptr<opentelemetry::ext::http::client::EventHandler> callback) noexcept
 {
-  curl_operation_->Abort();
+  is_session_active_.store(true, std::memory_order_release);
+  std::string url       = host_ + std::string(http_request_->uri_);
+  auto callback_ptr     = callback.get();
+  bool reuse_connection = false;
+  if (http_client_.GetMaxSessionsPerConnection() > 0)
+  {
+    reuse_connection = session_id_ % http_client_.GetMaxSessionsPerConnection() != 0;
```

Comment:
Sorry, this logic is not very clear to me. 

If (say) max_session_per_connection = 8, `reuse_connection` would be true for session_id_connection between 0 - 7, false for 8, and then again true for 9 - 15. As session_id keep incrementing for each session. Are we somehow reusing single curl connection for session_id 0 to 7, and then session_id 9 to 15..and so one ?

Reply:
> Sorry, this logic is not very clear to me.
> 
> If (say) max_session_per_connection = 8, `reuse_connection` would be true for session_id_connection between 0 - 7, false for 8, and then again true for 9 - 15. As session_id keep incrementing for each session. Are we somehow reusing single curl connection for session_id 0 to 7, and then session_id 9 to 15..and so one ?

`reuse_connection` would be `false` for session_id 0, and would be true for 1-7. And then `false` for 8 and `true` for `9-15`, and so on. It set `reuse_connection` to false for every 8 sessions. and we will tell `libcurl` do not reuse and create a new connection(socket) to handle requests.By this way, we can make a lot HTTP requests concurrently at one time(Requests in the same connection are still be sent one by one).